src/coreComponents/physicsSolvers/contact/SolidMechanicsLagrangeContact.cpp

```diff
@@ -2329,6 +2350,90 @@ bool SolidMechanicsLagrangeContact::updateConfiguration( DomainPartition & domai
   return changedArea <= m_nonlinearSolverParameters.m_configurationTolerance * totalArea;
 }
 
+void SolidMechanicsLagrangeContact::tryLocalYieldAcceleration( integer const configurationLoopIter,
+                                                               localIndex const kfe,
+                                                               real64 const currentTau_unscaled,
+                                                               real64 const limitTau,
+                                                               real64 const currentTau,
+                                                               integer & fractureState )
+{
+  using namespace fields::contact;
+
+  if( configurationLoopIter == 0 )
+  {
+    m_x0[kfe] = currentTau_unscaled - limitTau;
+  }
```

Comment:
I tried the example suggested by Matteo Frigo ([ContactMechanics_SimpleCubes_smoke](https://github.com/GEOS-DEV/GEOS/blob/develop/inputFiles/lagrangianContactMechanics/ContactMechanics_SimpleCubes_smoke.xml)).

It happens that `total_size` computed in `initializeAccelerationVariables` is equal to 0, hence GEOS crashes here because `m_x0` has zero size, as all the other allocated arrays needed for the nonlinear acceleration.


 I suspect it is because in that example the `SurfaceGenerator` is used to split the mesh. @sohailwaziri: please make sure `initializeAccelerationVariables` is called after the mesh has been split if the `SurfaceGenerator` is used. Thanks

Reply:
> I tried the example suggested by @matteofrigo5 ([ContactMechanics_SimpleCubes_smoke](https://github.com/GEOS-DEV/GEOS/blob/develop/inputFiles/lagrangianContactMechanics/ContactMechanics_SimpleCubes_smoke.xml)).
> 
> It happens that `total_size` computed in `initializeAccelerationVariables` is equal to 0, hence GEOS crashes here because `m_x0` has zero size, as all the other allocated arrays needed for the nonlinear acceleration.
> 
> I suspect it is because in that example the `SurfaceGenerator` is used to split the mesh.
> 
> @sohailwaziri: please make sure `initializeAccelerationVariables` is called after the mesh has been split, when the `SurfaceGenerator` is used. Thanks!

i got it fixed by moving `initializeAccelerationVariables` into `setupSystem`

without acceleration:
```
lagrangiancontact, number of time steps: 11
lagrangiancontact, number of successful outer loop iterations: 13
lagrangiancontact, number of successful nonlinear iterations: 94
lagrangiancontact, number of successful linear iterations: 94
lagrangiancontact, number of time step cuts: 1
lagrangiancontact, number of discarded outer loop iterations: 5
lagrangiancontact, number of discarded nonlinear iterations: 35
lagrangiancontact, number of discarded linear iterations: 35
```

with acceleration:
```
lagrangiancontact, number of time steps: 11
lagrangiancontact, number of successful outer loop iterations: 13
lagrangiancontact, number of successful nonlinear iterations: 95
lagrangiancontact, number of successful linear iterations: 95
lagrangiancontact, number of time step cuts: 1
lagrangiancontact, number of discarded outer loop iterations: 5
lagrangiancontact, number of discarded nonlinear iterations: 35
lagrangiancontact, number of discarded linear iterations: 35
```

so acceleration does not hurt but does not help either
maybe some parameter needs to be tuned as Sohail mentioned yesterday